skfda/misc/operators/_linear_differential_operator.py

```diff
@@ -577,6 +579,54 @@ def bspline_penalty_matrix_optimized(
     return penalty_matrix
 
 
+def _optimized_operator_evaluation_in_grid(
+    linear_operator: LinearDifferentialOperator,
+    grid_points: NDArrayFloat,
+    findiff_accuracy: int,
+) -> NDArrayFloat:
+    """
+    Compute the linear operator applied to the delta basis of the grid.
+
+    The delta basis is the basis where the i-th basis function is 1 in the
+    i-th point and 0 in the rest of points. The matrix returned by this
+    function is the matrix where the i-th row contians the values of the
+    linear operator applied to the i-th basis function of the delta basis.
+    """
+    n_points = grid_points.shape[0]
+    diff_coefficients = np.zeros((n_points, n_points))
+
+    # The desired matrix can be obtained by appending the coefficients
+    # of the finite differences for each point column-wise.
+    for dif_order, w in enumerate(linear_operator.weights):
+        if w == 0:
+            continue
+        for point_index in range(n_points):
```

Comment:
It is not slow. On the contrary, this whole function only takes around 3% of the runtime spent calculating the penalty matrix. I understand loops usually indicate slow code, but, in this case, having a linear algorithm outweighs this issue. 
Regarding the vectorization, I did look for it in the documentation but to no avail. Moreover, since there are no performance issues, I do not see a need for it. They are also unlikely to add it. It is already possible to convert a differential operator (as an object in findiff) to a matrix, so they would probably point to that. However, I tested it, and it was considerably slower.

Reply:
This is the relevant portion of the call tree in the profiling output. The whole running time corresponds to executing the last test case of the FPCA regularization test but with a grid of 5000 points. As you can see, the running time of this function amounts to a negligible portion of the total running time.
![image](https://user-images.githubusercontent.com/44179156/223199328-1299fdd1-c568-482c-bf12-31135bdb65f1.png)